jetty9/src/com/thoughtworks/go/server/config/ConfigurableSSLSettings.java

```diff
@@ -0,0 +1,187 @@
+/*************************GO-LICENSE-START*********************************
+ * Copyright 2015 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *************************GO-LICENSE-END***********************************/
+
+package com.thoughtworks.go.server.config;
+
+import com.google.gson.Gson;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+import com.thoughtworks.go.util.StringUtil;
+import com.thoughtworks.go.util.SystemEnvironment;
+import org.apache.commons.io.IOUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class ConfigurableSSLSettings implements SSLConfig {
+    private final Config config;
+
+    public ConfigurableSSLSettings(SystemEnvironment systemEnvironment) {
+        try {
+            String systemConfiguredSslConfigFile = systemEnvironment.get(SystemEnvironment.GO_SSL_CONFIG_FILE_PATH);
+            InputStream sslConfigStream = getClass().getResourceAsStream(systemConfiguredSslConfigFile);
+            String json = IOUtils.toString(sslConfigStream);
+            config = new Gson().fromJson(json, Config.class).setDefaults();
+
+            if (!StringUtil.isBlank(systemEnvironment.get(SystemEnvironment.USER_CONFIGURED_SSL_CONFIG_FILE_PATH))) {
+                File userConfiguredSslConfigFile = new File(systemEnvironment.configDir(), systemEnvironment.get(SystemEnvironment.USER_CONFIGURED_SSL_CONFIG_FILE_PATH));
+                if (userConfiguredSslConfigFile.exists()) {
+                    Config userDefinedConfig = new Gson().fromJson(IOUtils.toString(new FileInputStream(userConfiguredSslConfigFile)), Config.class);
```

Comment:
I prefer a flat style of options over the verbose way of config to make things simpler.
```
ssl.config.ciphers.include = STRONG_CIPHER
ssl.config.ciphers.exclude = WEAK_CIPHER
ssl.config.protocols.include = STRONG_PROTOCOL
ssl.config.protocols.excludes = WEAK_PROTOCOL
```
Not very sure on system properties or file however.
Also aren't these value set on startup. So even if the files synced don't we have to restart for the server to pick up these properties?